### PR TITLE
Fixes RPD aggression

### LINKED ISSUE #23714: Painting with the RPD makes you hit the pipes
ABDUCTOR has hit the pipe with Rapid Piping Device (RPD)!
You paint the pipe yellow.
ABDUCTOR has hit the pipe with Rapid Piping Device (RPD)!
You paint the pipe yellow.
ABDUCTOR has hit the pipe with Rapid Piping Device (RPD)!
You paint the pipe yellow.

### PR DESCRIPTION
**Pull request by Cyberboss**
Fixes #23714

:cl: Cyberboss
fix: Pipe painters are no longer aggressive
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/RPD.dm [hunk 1]
```diff
@@ -518,50 +520,62 @@ var/global/list/RPD_recipes=list(
...
+				return
 
 			// Avoid spewing errors about invalid mode -1 when clicking on stuff that aren't pipes.
 			user << "<span class='warning'>The [src]'s error light flickers!  Perhaps you need to only use it on pipes and pipe meters?</span>"
-			return 0
+			return
+		else
+			return ..()
+
+/obj/item/weapon/pipe_dispenser/attack(atom/A, mob/user)
```
Comment (ChangelingRain) [inline code/game/objects/items/weapons/RPD.dm]: I'm not sure what you're doing but I think you've fucked up the click chain really bad.

File: code/game/objects/items/weapons/RPD.dm [hunk 2]
```diff
@@ -518,50 +520,62 @@ var/global/list/RPD_recipes=list(
 		show_menu(usr)
 
 
-/obj/item/weapon/pipe_dispenser/afterattack(atom/A, mob/user)
```
Comment (ChangelingRain) [inline code/game/objects/items/weapons/RPD.dm]: this should probably use pre_attackby, with appropriate return values

File: code/game/objects/items/weapons/RPD.dm [hunk 3]
```diff
@@ -518,50 +520,62 @@ var/global/list/RPD_recipes=list(
...
+
+/obj/item/weapon/pipe_dispenser/attack(atom/A, mob/user)
+	if(!in_range(A,user) || loc != user)
+		return 0
+
+	if(!user.IsAdvancedToolUser())
+		user << "<span class='warning'>You don't have the dexterity to do this!</span>"
+		return 0
+
+	if(istype(A,/area/shuttle)||istype(A,/turf/open/space/transit))
```
Comment (ChangelingRain) [inline code/game/objects/items/weapons/RPD.dm]: what the fuck. why are you checking if something you've clicked on is an area. that should never happen.
I know this was here before, but what the fuck, why is it still here?

File: code/game/objects/items/weapons/RPD.dm (hunk 1 above)
Comment (ChangelingRain) [inline code/game/objects/items/weapons/RPD.dm]: yeah this is the proc for hitting a mob with this item, you have some critical errors here

File: code/game/objects/items/weapons/RPD.dm (hunk 2 above)
Comment (Cyberboss) [inline code/game/objects/items/weapons/RPD.dm]: pre_attackby isn't ideal for the uses that need turfs

Comment (ChangelingRain) [inline code/game/objects/items/weapons/RPD.dm]: ... isn't ideal how? it takes turfs as arguments just fine, same as afterattack.

File: code/game/objects/items/weapons/RPD.dm [hunk 4]
```diff
@@ -518,50 +520,52 @@ var/global/list/RPD_recipes=list(
...
-
-	if(istype(A,/area/shuttle)||istype(A,/turf/open/space/transit))
-		return 0
-
-	//So that changing the menu settings doesn't affect the pipes already being built.
-	var/queued_p_type = p_type
-	var/queued_p_dir = p_dir
-	var/queued_p_flipped = p_flipped
+/obj/item/weapon/pipe_dispenser/pre_attackby(obj/A, mob/user)
+	if(!in_range(A,user) || loc != user || !user.IsAdvancedToolUser())
```
Comment (ChangelingRain) [inline code/game/objects/items/weapons/RPD.dm]: the first two are, uh, just anti-TK checks for some reason?

File: code/game/objects/items/weapons/RPD.dm (hunk 2 above)
Comment (Cyberboss) [inline code/game/objects/items/weapons/RPD.dm]: See #23727

**Pull request merged**